brave/brave-ios
# Fix #5731: Transaction Parser Refactor

## Summary of Changes
- Refactored `TransactionParser` to generate a new `ParsedTransaction` model. `ParsedTransaction` model contains common properties like `fromAddress`, `toAddress` and named accounts and the network symbol. Additionally it holds an enum for specific transaction type details.

This pull request fixes #5731

## Submitter Checklist:


## Test Plan:

1. Create a few transactions on `development` branch:
    1. ETH send
    2. DAI send (or any non-ETH token)
    3. ERC 20 Approval
    4. Swap transaction ETH -> any non-ETH token
    5. Swap transaction from non-ETH token to a different non-ETH token
2. Verify display of these transactions in the transaction lists on Asset Detail, Account Detail and transaction history (from wallet panel). May help to take a screenshot for comparison.
3. Switch to `transaction-parser` branch
4. Verify display of the same transactions again, comparing against values from step 2.


## Screenshots:

![transactions on development](https://user-images.githubusercontent.com/5314553/180329198-a1ac2a94-fc53-404e-b516-8486cea0699c.png) | ![updated parsed transactions](https://user-images.githubusercontent.com/5314553/180329203-a5ccb3d3-f3e0-4f30-9a7e-8a1a98a01c25.png)
--|--


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
